src/views/pay.js

```diff
@@ -34,19 +37,29 @@ class Pay extends Component {
           onChangeText={payment => {
             actionsPayments
               .decodePaymentRequest(payment)
-              .then(num_satoshis => this.setState({ amount: num_satoshis }))
-              .catch(() => {});
+              .then(res => {
+                this.setState({
+                  paymentRequest: true,
+                  amount: res.num_satoshis,
+                  description: res.description,
+                });
```

Comment:
In general we should be using the mobx store instead of component specific state (see channels view). I haven't had a chance to go though all components yet. @kevinejohn is there a reason state is being used in some places and the mobx store in others?

Reply:
I will leave it as is for now. If you want me to change it to use mobx, lmk or it could probably be a separate PR. Have never used it before, but could play around with it.

Reply:
In general I only use mobx for shared global state and setState for anything local to a specific component. It keeps things more modular.

Reply:
The current tests strategy is to unit test the mobx actions, computed logic which let‘s us test the store as well. I‘d prefer to keep the react components purely functional and stateless if possible to improve testability and consistency. So I‘d prefer to move state changes to actions/computed code.

Reply:
That sounds good to me.

Reply:
👍🏼